### PR TITLE
feat: multi-level topics selection impl exlm-1012

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1012

Test URLs:

Before: https://main--exlm--adobe-experience-league.hlx.live/en/browse/experience-manager
After: https://feature-exlm-1012--exlm--adobe-experience-league.hlx.live/en/browse/experience-manager

### CODE REVIEW

File: blocks/browse-filters/browse-topics.js [hunk 1]
```diff
@@ -29,6 +29,18 @@ export function formattedTags(inputString) {
...
+  if (!product) {
+    return `@el_features="${type}"`;
+  }
+  if (!version) {
+    return `@el_product="${product}`;
+  }
+  const isSolutionType = type.toLowerCase().includes('solution');
+  return `(@el_product="${product}" AND @el_${isSolutionType ? 'solution' : 'features'}="${version}")`;
+};
+
```
Comment (noormd08) [inline blocks/browse-filters/browse-topics.js:43]: multiple products with same values are considered?

File: blocks/browse-filters/browse-filter-utils.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ export const getSelectedTopics = (filterInfo) => {
   if (!filterInfo) {
     return [];
   }
-  const featuresCheck = filterInfo.match(/@el_features=("[^"]*")/g) || [];
+  const featuresCheck = filterInfo.match(/@el_product=("[^"]*")/g) ?? filterInfo.match(/@el_features=("[^"]*")/g) ?? [];
```
Comment (noormd08) [inline blocks/browse-filters/browse-filter-utils.js:125]: Try avoiding regEx

File: blocks/browse-filters/browse-topics.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply blocks/browse-filters/browse-topics.js:43]: yes Noor.

File: blocks/browse-filters/browse-filter-utils.js (hunk 2 above)
Comment (nitin-rachabathuni) [reply blocks/browse-filters/browse-filter-utils.js:125]: @noormd08 unfortunately we are not able to regex here.

**Review by noormd08 — APPROVED** [review]

**Pull request merged**